client/client.go

```diff
@@ -253,24 +253,31 @@ func NewClientWithAddress(address string) (client Client, err error) {
 // NewClientWithAddressContext instantiates Dapr using specific address (including port).
 // Uses the provided context to create the connection.
 func NewClientWithAddressContext(ctx context.Context, address string) (client Client, err error) {
-	if address == "" {
-		return nil, errors.New("empty address")
-	}
-	logger.Printf("dapr client initializing for: %s", address)
-
 	timeoutSeconds, err := getClientTimeoutSeconds()
 	if err != nil {
 		return nil, err
 	}
 	ctx, cancel := context.WithTimeout(ctx, time.Duration(timeoutSeconds)*time.Second)
+	defer cancel()
+	return NewClientWithAddressContextNoEnvTimeout(ctx, address)
+}
+
+// NewClientWithAddressContextNoEnvTimeout instantiates Dapr using specific address (including port).
+// Ignores the DAPR_GRPC_TIMEOUT_SECONDS environment variable and has no
+// default timeout.
+func NewClientWithAddressContextNoEnvTimeout(ctx context.Context, address string) (client Client, err error) {
```

Comment:
Recommend changing to `NewClientWithAddressContextNoTimeout`

Reply:
Done.